File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    X = ''
    Xcount = 0
    another = 0
    for i in range(len(s)):
        if Xcount == 0:
            X=s[i]
            Xcount+=1
            answer+=1
            continue

        if X == s[i]:
            Xcount+=1
        else:
            another+=1

        if Xcount == another:
            Xcount=0
            another=0
    return answer

File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let answer = 0, diff_count = 0, same_count = 0
    let character = ''
    
    for(c of s){
        if(character === '') character = c;
        
        (c === character) ? same_count++ : diff_count++
        
        if(diff_count === same_count){
            answer++
            character = ''
            diff_count = 0
            same_count = 0
        }
    }
    
    if(diff_count + same_count) answer++;
    return answer
}